sql/hive/src/main/scala/org/apache/spark/sql/hive/prestoviews/PrestoViewDecoder.scala

```diff
@@ -0,0 +1,68 @@
+
+package org.apache.spark.sql.hive.prestoviews
+
+import java.nio.charset.StandardCharsets
+
+import com.google.common.base.Preconditions
+import com.google.gson.Gson
+import org.apache.hadoop.hive.metastore.api.FieldSchema
+import org.apache.spark.sql.types.StructType
+
+case class DecodedPrestoViewObject(originalSql: String,
+                                   catalog: String,
+                                   schema: String,
+                                   columns: Array[ColumnDefinition])
+
+case class ColumnDefinition(name: String, `type`: String)
+
+object PrestoViewDecodeParser {
+
+  import org.apache.spark.sql.hive.client.HiveClientImpl._
+  private val dataTypeMap = Map("varchar"-> "String")
+  private val GSON = new Gson()
+  private val VIEW_PREFIX = "/* Presto View: "
+  private val VIEW_PREFIX_MARKER = "/* Presto View */"
+  private val VIEW_SUFFIX = " */"
+  def isPrestoView(sqlText: String): Boolean = {
+    val result = sqlText.startsWith(VIEW_PREFIX)
+    result
+  }
+
+  def isViewMarker(sqlText: String): Boolean = {
+    sqlText.startsWith(VIEW_PREFIX_MARKER)
+  }
+
+  def decode(sqlText: String): (String, StructType) = {
+    val result = decodeViewData(sqlText)
+    val prestoViewJson = GSON.fromJson(result, classOf[DecodedPrestoViewObject])
```

Comment:
some debug logging of the extracted view here maybe?

Reply:
Sure will add proper logs while doing the actual fix. This PR was just to share high-level logic.